operator/tests/e2e-with-flags/nodepools-delete/01-add-second-pool.yaml

```diff
@@ -8,6 +8,8 @@ spec:
   nodePools:
     - name: first
       replicas: 3
+      storage: {}
```

Comment:
These changes seem odd. I get that they were blocking this passing CI all of a sudden (wondering if we have something wrong with the tests/CI that is making them occasionally pass even without the change?), so I'm fine with adding these as a fix for now... that said, @birdayz is there a reason why these are marked with `required`? It looks like all of the corresponding struct fields are optional...

https://github.com/redpanda-data/redpanda-operator/blob/0312468ffee31d38ff1882ffa4143177c10cc09f/operator/api/vectorized/v1alpha1/cluster_types.go#L213-L215

Any chance that could be changed to `omitempty`? No need to change to a pointer or anything, but I don't like the idea of hacking in an empty json map just for the sake of passing field validation.